app/_components/toggle-switch/ToggleSwitch.tsx

```diff
@@ -0,0 +1,37 @@
+'use client';
+
+import { useState } from 'react';
+
+type ToggleSwitchPropsType = {
+  getToggleValue: (toggle: boolean) => void;
+};
+
+const ToggleSwitch = ({ getToggleValue }: ToggleSwitchPropsType) => {
+  const [isOn, setIsOn] = useState(false);
+
+  const handleToggleSwitch = () => {
+    setIsOn(prev => !prev);
+    getToggleValue(isOn);
+  };
```

Comment:
이렇게 하면 아마 `isOn`이 이전 값을 출력할 것 같습니다..!
`batching`과 관련된 부분으로 저는 이해하는데요.

이렇게 해보면 어떨까 싶네요..! 🧐

```suggestion
  const handleToggleSwitch = () => {
    setIsOn(prev => !prev);
  };

  getToggleValue(isOn);
```

Reply:
테스트해보고 수정하겠습니다.